crates/core/src/instance.rs

```diff
@@ -297,6 +316,11 @@ impl Instance {
                 while kill_receiver.try_recv().is_err() {
                     log_trace!(context, "Checking holding queue...");
                     loop {
+                        // TODO: TRACING: it would be ideal to be able to associate a tracing Span with each queued holding workflow.
+                        // To do this, we'd need to store a Span in each item of the DhtStore::queued_holding_workflows.
+                        // However, Span is not Clone, and the entire DhtStore needs to be Cloned.
+                        // So, the span has to be cut short here until we stop cloning the state.
+
```

Comment:
Again, what about putting `Arc` wrapped spans into the state? We are doing this with the state slices already to not clone the whole state every time - just the root reference.

Reply:
@lucksus The problem is that we rely on the Span being dropped to mark the end of the span and to actually send it for reporting. If we allow it inside an `Arc`, we lose control over when it goes out of scope if we don't keep track of every clone. If all clones don't drop at the same time, we get an inaccurate span. We could solve this with explicit coordination, but that's just more complexity. Unless you see a simple way that I'm not seeing?

Reply:
I see. Well, that explicit coordination could just be an `Arc<Option<..>>`, right?
(Or `Arc<Mutex<Option<..>>>`)